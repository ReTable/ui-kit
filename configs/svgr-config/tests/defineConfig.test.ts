import { transform } from '@svgr/core';
import { describe, expect, it } from 'vitest';

import { Options, defineConfig } from '~';

const icon = `
  <svg xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink">
    <rect x="10" y="10" height="100" width="100"
      style="stroke:#ff0000; fill: #0000ff"/>
  </svg>
`;

async function transformIcon(options: Options) {
  const config = { ...defineConfig(options), plugins: ['@svgr/plugin-jsx'] };

  return transform(icon, config, {
    componentName: 'SvgIcon',
  });
}

describe('defineConfig', () => {
  it('add scope and displayName without name transformation', async () => {
    const component = await transformIcon({ scope: 'svgr-config' });

    expect(component).toMatchSnapshot();
  });

  it('add scope and displayName with name transformation', async () => {
    const component = await transformIcon({
      scope: 'svgr-config',

      transformName(name) {
        return `Ui${name.slice(3)}`;
      },
    });

    expect(component).toMatchSnapshot();
  });
});
