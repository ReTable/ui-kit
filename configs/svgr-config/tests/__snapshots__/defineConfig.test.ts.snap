// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defineConfig > add scope and displayName with name transformation 1`] = `
"import * as React from "react";
import { memo } from "react";
const UiIcon = props => <svg xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" {...props}><rect x={10} y={10} height={100} width={100} style={{
    stroke: "#ff0000",
    fill: "#0000ff"
  }} /></svg>;
const Memo = memo(UiIcon);
export default Memo;
Memo.displayName = "svgr-config(UiIcon)";"
`;

exports[`defineConfig > add scope and displayName without name transformation 1`] = `
"import * as React from "react";
import { memo } from "react";
const SvgIcon = props => <svg xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink" {...props}><rect x={10} y={10} height={100} width={100} style={{
    stroke: "#ff0000",
    fill: "#0000ff"
  }} /></svg>;
const Memo = memo(SvgIcon);
export default Memo;
Memo.displayName = "svgr-config(SvgIcon)";"
`;
